Dell/recovery_backend.py

```diff
@@ -777,6 +777,31 @@ def validate_driver_package(self, package, sender=None, conn=None):
         logging.debug("Validation complete: valid %s" % valid)
         self.report_package_info(valid, description, error_warning)
 
+    @dbus.service.method(DBUS_INTERFACE_NAME,
+        in_signature = 'b', out_signature = '', sender_keyword = 'sender',
+        connection_keyword = 'conn')
+    #The funcution is used to recovery DHC
+    def enable_boot_to_restore_dhc(self, reboot, sender=None, conn=None):
+        """Enables the default one-time boot option to be recovery"""
+        self._reset_timeout()
+        recovery=find_partition().decode('utf-8')
+        mntdir=self.request_mount(recovery, 'rw', sender,conn)
+        #Create dhc_flag to provide judgment for recovery DHC to delete copied dhc file, and dhc_flag will be deleted after recovery DHC.  
+        os.makedirs(os.path.join(mntdir, "dhc_flag"))
+        #Delete seedfile for copy DHC seed file
+        if os.path.exists(os.path.join(mntdir, "preseed", "dell-recovery.seed")):
+            os.remove(os.path.join(mntdir,"preseed", "dell-recovery.seed"))
+        if not os.path.exists(os.path.join(mntdir, "scripts")):
+            os.makedir(os.path.join(mntdir, "scripts"))
+        #copy DHC file to corresponding dir for recovery DHC
+        dirs=["casper", "preseed", "scripts"]
```

Comment:
So if i'm following this correctly - you're duplicating everything from `/mnt/dhc/*` to `/mnt/`, right?

Why not instead just reference contained items in place?  It seems like a waste of space on the recovery partition to have to duplicate all of these.

Reply:
Only the scripts and seed files written for DHC are copied. For DHC package and slides, we will not copy them, but read the corresponding path through the script.

Reply:
Where possible, it's better to remove state from the equation.  Go off the presence of a single flag file to figure out which files to load rather than needing to copy pieces around.  It's much more difficult to debug problems that only crop up after a recovery or a recovery from a recovery when items have been copied around and the state isn't guaranteed identical.

You already are creating a `dhc_flag`, I would propose look for that and when it's there search for additional locations for these items.

Reply:
Yes, If using dhc_flag can search additional location, that's fine. If not,  is it possible to use the rename method, like efi directory.